base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -452,7 +447,22 @@ where B: BlockchainBackend + 'static
             );
             return Ok(());
         }
-
+        {
+            // we have this in mutex to ensure we dont have any funnies with rwlock with multiples requests first
+            // passing the check then all wanting to reconcile. We should only have 1 request passing
+            let mut lock = self.list_of_reconciling_blocks.lock().await;
+            let list_of_reconciling_blocks = lock.deref_mut();
+            if list_of_reconciling_blocks.contains(&block_hash) {
+                debug!(
+                    target: LOG_TARGET,
+                    "Block with hash `{}` is already being reconciled",
+                    block_hash.to_hex()
+                );
+                return Ok(());
+            } else {
+                list_of_reconciling_blocks.insert(block_hash);
+            }
```

Comment:
nit: there is no need to explicitly dereference a lock guard (you may want to rename lock to list_of_reconciling_blocks for clarity)
```suggestion
          
            if list_of_reconciling_blocks.contains(&block_hash) {
                debug!(
                    target: LOG_TARGET,
                    "Block with hash `{}` is already being reconciled",
                    block_hash.to_hex()
                );
                return Ok(());
            } else {
                list_of_reconciling_blocks.insert(block_hash);
            }
```